### PR TITLE
Switch to MkDocs

### PR DESCRIPTION
**Pull request by cbrnr**
This is the first try, let's see if everything works (especially on RTD).

### CODE REVIEW

**Comment by cbrnr** [comment]
@hofaflo WDYT?

**Comment by hofaflo** [comment]
Looks good! There are a few things the sphinx version has which I'd like to keep:
- the [API overview page](https://sleepecg.readthedocs.io/en/stable/api.html) with short descriptions of all functions and classes
- there's an individual page for each function
- the [source] links

**Comment by cbrnr** [comment]
> the [API overview page](https://sleepecg.readthedocs.io/en/stable/api.html) with short descriptions of all functions and classes

This needs to be created manually. Unfortunately, the API section in the left sidebar cannot be associated with a page, so I don't know how to best include an overview. Do you have an idea?

> there's an individual page for each function

Why do you want individual pages?

> the [source] links

The source code is included at the end of every documented API object. Is this OK? I am not sure if linking to GitHub works.

**Comment by hofaflo** [comment]
> > the [API overview page](https://sleepecg.readthedocs.io/en/stable/api.html) with short descriptions of all functions and classes
> 
> This needs to be created manually. Unfortunately, the API section in the left sidebar cannot be associated with a page, so I don't know how to best include an overview. Do you have an idea?

Attaching a page to a navigation section is possible by using [section index pages](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages). I'll look into filling the page with the first docstring lines automatically, maybe in another PR?


> > there's an individual page for each function
> 
> Why do you want individual pages?

Personal preference, I find it less crowded... but if it's hard to do I'm fine with combined pages as well.

> > the [source] links
> 
> The source code is included at the end of every documented API object. Is this OK? Otherwise, https://ai2business.github.io/mkdocstrings-sourcelink/ looks like it does what you want.

I guess it's okay for now, I'll try to add GitHub links later.

**Comment by cbrnr** [comment]
A section index page doesn't seem to be too difficult to implement. GitHub links are much more complicated, as they require a custom build pipeline (plain old `mkdocs build` doesn't cut it anymore).

But I agree we can always add new stuff in follow-up PRs, this one is already pretty large and most things already work (indeed, many things work better like Markdown support in docstrings).

So OK to merge this one after I've added a changelog entry?

**Comment by hofaflo** [comment]
> So OK to merge this one after I've added a changelog entry?

Almost, a few more things:

- [ ] not all API pages are (fully) populated yet
- [ ] `doc/source/generated/` can be removed from `.gitignore`
- [ ] I think the `[doc]` dependencies should be part of `[dev]` as well

**Comment by cbrnr** [comment]
Looking good – @hofaflo feel free to merge!

File: docs/classification.md [hunk 1]
```diff
@@ -30,13 +29,14 @@ A weaker weighting approach is likely required to find the optimal middle ground
 You are very welcome to add new classifiers to SleepECG!
 To do so, please submit a [pull request](https://github.com/cbrnr/sleepecg/pulls).
 Make sure it contains the following elements:
+
 - The script required to replicate and validate the classifer (placed in `examples/classifiers`).
-- The classifier itself (saved as a `.zip` file with `save_classifier`) placed in `sleepecg/classifiers`.
+- The classifier itself (saved as a `.zip` file with [`sleepecg.save_classifier()`][sleepecg.save_classifier]) placed in `sleepecg/classifiers`.
```
Comment (hofaflo) [inline docs/classification.md:34]:
```suggestion
- The classifier itself (saved as a `.zip` file with [`sleepecg.save_classifier()`](sleepecg.save_classifier) placed in `sleepecg/classifiers`.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -1,7 +1,6 @@
-(configuration)=
 # Configuration
 SleepECG provides an interface to set and persist global configuration values. See
-{func}`sleepecg.get_config` and {func}`sleepecg.set_config` for usage instructions.
+[`sleepecg.get_config()`][sleepecg.get_config] and [`sleepecg.set_config()`][sleepecg.set_config] for usage instructions.
```
Comment (hofaflo) [inline docs/configuration.md:3]:
```suggestion
[`sleepecg.get_config()`](sleepecg.get_config) and [`sleepecg.set_config()`](sleepecg.set_config) for usage instructions.
```

**Comment by hofaflo** [comment]
Thanks @cbrnr!

**Pull request merged**